### ISSUE TITLE
Sticky header content overflows header

### ISSUE THREAD

**Issue by caugner** [mdn/yari#6069]
### Summary

The header overflows with certain viewport widths, overlapping with the breadcrumbs and content.

### URL

https://developer.allizom.org/en-US/docs/Web/HTML/Element/dialog

### Reproduction steps

1. Open https://developer.allizom.org/en-US/docs/Web/HTML/Element/dialog
2. Resize the window to about 800px in width.

### Expected behavior

Two options:

1. The header should collapse into the hamburger menu, as soon as it no longer fits.
2. The header should increase in height, if it no longer fits on one line.

The problem of 2 is that the sticky header would significantly more space that way.

### Actual behavior

The header splits into two lines, but the second line overflows, not increasing the header height, and therefore overlaps with the page elements below.

### Device

Desktop

### Browser

Firefox

### Browser version

Pre-release (e.g. Beta, Nightly or Canary)

### Operating system

Mac OS

### Screenshot

<img width="822" alt="image" src="https://user-images.githubusercontent.com/495429/165273703-a5814863-1789-4f55-89dc-2db8d84eb4de.png">

### Anything else?

Originally reported by @fiji-flo on [Slack](https://mozilla.slack.com/archives/G018PLBEUES/p1650964844424989) (internal).

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mdn/yari/discussions).

**Comment by caugner** [comment]
Git bisect has concluded that this issue was introduced in c9a73a08010ab641efb73a8eb56901b96c277aaa (i.e. https://github.com/mdn/yari/pull/5710).

```
c9a73a08010ab641efb73a8eb56901b96c277aaa is the first bad commit
commit c9a73a08010ab641efb73a8eb56901b96c277aaa
Author: Schalk Neethling <sneethling@mozilla.com>
Date:   Tue Apr 19 23:12:23 2022 +0200

    fix(a11y): improve accessibility, readability and usability (#5710)
```

**Comment by caugner** [comment]
And here's the culprit:

https://github.com/mdn/yari/blob/8ff53ddde455623c77113945695dc010a07c7f13/client/src/ui/organisms/top-navigation/index.scss#L3-L4

**Comment by caugner** [comment]
Fixed by #6106.
